addons/payment_stripe/models/payment.py

```diff
@@ -40,16 +40,16 @@ def stripe_form_generate_values(self, tx_values):
         stripe_tx_values = dict(tx_values)
         temp_stripe_tx_values = {
             'company': self.company_id.name,
-            'amount': tx_values.get('amount'),
-            'currency': tx_values.get('currency') and tx_values.get('currency').name or '',
-            'currency_id': tx_values.get('currency') and tx_values.get('currency').id or '',
-            'address_line1': tx_values['partner_address'],
-            'address_city': tx_values['partner_city'],
-            'address_country': tx_values['partner_country'] and tx_values['partner_country'].name or '',
-            'email': tx_values['partner_email'],
-            'address_zip': tx_values['partner_zip'],
-            'name': tx_values['partner_name'],
-            'phone': tx_values['partner_phone'],
+            'amount': tx_values.get('amount'),  # Mandatory
```

Comment:
If it is mandatory why not using brackets, as it looks like you are changing them :p ?